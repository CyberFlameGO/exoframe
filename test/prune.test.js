/* eslint-env jest */
// mock config for testing
jest.mock('../src/config', () => require('./__mocks__/config'));

// npm packages
const nock = require('nock');
const sinon = require('sinon');

// our packages
const {handler: system} = require('../src/commands/system');

// test update
test('Should execute prune', done => {
  // handle correct request
  const pruneServer = nock('http://localhost:8080')
    .post('/system/prune')
    .reply(200, {pruned: true, data: [{SpaceReclaimed: 1024}]});
  // spy on console
  const consoleSpy = sinon.spy(console, 'log');
  // execute login
  system({cmd: 'prune'}).then(() => {
    // make sure log in was successful
    // check that server was called
    expect(pruneServer.isDone()).toBeTruthy();
    // first check console output
    expect(consoleSpy.args).toMatchSnapshot();
    // restore console
    console.log.restore();
    pruneServer.done();
    done();
  });
});
