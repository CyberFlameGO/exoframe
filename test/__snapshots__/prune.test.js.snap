// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should execute prune 1`] = `
Array [
  Array [
    "Data prune successful!",
  ],
  Array [
    "",
  ],
  Array [
    "Reclaimed:",
    "1.02 kB",
  ],
]
`;
